srcdeps-core/src/main/java/org/l2x6/srcdeps/core/fs/PathLocker.java

```diff
@@ -36,18 +36,55 @@
  * A simple facility to guarantee both thread level and OS process level exclusive access to a filesystem path.
  *
  * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ * @param <M>
+ *            metadata describing the path to lock
  */
 @Named
 @Singleton
-public class PathLocker {
+public class PathLocker<M> {
+
+    /**
+     * A pair consisting of a {@link ReentrantLock} and a metadata instance.
+     *
+     * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+     *
+     * @param <M>
+     *            metadata describing the path to lock
+     */
+    private static class LockMetadataPair<M> {
```

Comment:
er... I admit I thought it does not matter much how such a private inner class will be called :)